pkg/cmd/taskrun/logs_test.go

```diff
@@ -0,0 +1,150 @@
+	"github.com/jonboulle/clockwork"
+	"github.com/knative/pkg/apis"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/logs"
+	tu "github.com/tektoncd/cli/pkg/test"
+	cb "github.com/tektoncd/cli/pkg/test/builder"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/test"
+	tb "github.com/tektoncd/pipeline/test/builder"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestLog_no_taskrun_arg(t *testing.T) {
+	c := Command(&tu.Params{})
+
+	_, err := tu.ExecuteCommand(c, "logs", "-n", "ns")
+	if err == nil {
+		t.Error("Expecting an error but it's empty")
+	}
+}
+
+func TestLog_missing_taskrun(t *testing.T) {
+	tr := []*v1alpha1.TaskRun{
+		tb.TaskRun("output-taskrun-1", "ns"),
+	}
+	cs, _ := test.SeedTestData(test.Data{TaskRuns: tr})
+	p := &tu.Params{Tekton: cs.Pipeline, Kube: cs.Kube}
+
+	c := Command(p)
+	got, _ := tu.ExecuteCommand(c, "logs", "output-taskrun-2", "-n", "ns")
+	expected := msgTRNotFoundErr + " : taskruns.tekton.dev \"output-taskrun-2\" not found \n"
+
+	if d := cmp.Diff(expected, got); d != "" {
+		t.Errorf("Unexpected output mismatch: \n%s\n", d)
+	}
+}
+
+func TestLog_valid_taskrun_logs(t *testing.T) {
+	var (
+		prstart      = clockwork.NewFakeClock()
+		ns           = "namespace"
+		task1Name    = "output-task"
+		tr1Name      = "output-task-1"
+		tr1StartTime = prstart.Now().Add(20 * time.Second)
+		tr1Pod       = "output-task-pod-123456"
+		tr1Step1Name = "writefile-step"
+	)
+	trs := []*v1alpha1.TaskRun{
+		tb.TaskRun(tr1Name, ns,
+			tb.TaskRunStatus(
+				tb.PodName(tr1Pod),
+				tb.TaskRunStartTime(tr1StartTime),
+				tb.Condition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+				tb.StepState(
+					cb.StepName(tr1Step1Name),
+					tb.StateTerminated(0),
+				),
+				tb.StepState(
+					cb.StepName("nop"),
+					tb.StateTerminated(0),
+				),
+			),
+			tb.TaskRunSpec(
+				tb.TaskRunTaskRef(task1Name),
+			),
+		),
+	}
+	pods := []*corev1.Pod{
+		tb.Pod(tr1Pod, ns,
+			tb.PodSpec(
+				tb.PodContainer(ContainerNameForStep(tr1Step1Name), tr1Step1Name+":latest"),
+				tb.PodContainer("nop", "override-with-nop:latest"),
+			),
+		),
+	}
+	fakeLogStream := logs.TaskRun(
+		logs.Task(task1Name, tr1Pod,
+			logs.Step(tr1Step1Name, ContainerNameForStep(tr1Step1Name),
+				"written a file",
+			),
+			logs.Step("nop", "nop",
+				"Build successful",
+			),
+		),
+	)
+
+	cs, _ := test.SeedTestData(test.Data{TaskRuns: trs, Pods: pods})
+	tr := fakeTaskRunLogs(task1Name, tr1Name, ns, cs)
+
+	output := fetchLogs(LogOpts(false), tr, logs.FakeLogFetcher(cs.Kube, fakeLogStream))
+	expectedLogs := []string{
+		"[output-task : writefile-step] written a file",
+		"[output-task : nop] Build successful",
+	}
+
+	expected := strings.Join(expectedLogs, "\n") + "\n"
+
+	if d := cmp.Diff(output, expected); d != "" {
+		t.Errorf("Unexpected output mismatch: \n%s\n", d)
+	}
+}
+
+func LogOpts(allSteps bool, tasks ...string) LogOptions {
```

Comment:
why is this exported?

Reply:
Do we really need this func?

Reply:
It should not be exported.
I tried removing it but this increases a few lines after adding a new test, it would be nice to keep it.